src/Parser/Monk/Brewmaster/Modules/Spells/RushingJadeWind.js

```diff
@@ -15,51 +14,17 @@ class RushingJadeWind extends Analyzer {
     combatants: Combatants,
   };
 
-  inactiveDuration = 0;
-  _lastDropped = 0;
-  _active = false;
-
   on_initialized() {
-    this._lastDropped = this.owner.fight.start_time;
     this.active = this.combatants.selected.hasTalent(SPELLS.RUSHING_JADE_WIND_TALENT.id);
   }
 
-  on_byPlayer_applybuff(event) {
-    if(event.ability.guid === RUSHING_JADE_WIND_BUFF.id) {
-      this.inactiveDuration += event.timestamp - this._lastDropped;
-      this._active = true;
-      debug && console.log("RJW re-applied", event.timestamp, this.inactiveDuration);
-    }
-  }
-
-  on_byPlayer_removebuff(event) {
-    if(event.ability.guid === RUSHING_JADE_WIND_BUFF.id) {
-      this._lastDropped = event.timestamp;
-      this._active = false;
-      debug && console.log("RJW dropped", event.timestamp);
-    }
-  }
-
-  on_finished() {
-    if(!this._active) {
-      this.inactiveDuration += this.owner.fight.end_time - this._lastDropped;
-    }
-  }
-
-  get uptime() {
-    if(this.owner.fightDuration === 0) {
-      return 0;
-    }
-    const downtime = this.inactiveDuration / this.owner.fightDuration;
-    return 1.0 - downtime;
-  }
-
   // using a suggestion rather than a checklist item for this as RJW is
   // purely offensive
   suggestions(when) {
-    when(this.uptime).isLessThan(0.95)
+    const uptime = this.combatants.selected.getBuffUptime(RUSHING_JADE_WIND_BUFF.id) / this.owner.fightDuration;
```

Comment:
For future reference it's generally best practice to make a getter for these checks so it can be used elsewhere too, it's easier to find and the suggestions method has less logic.